src/python/pants/option/global_options.py

```diff
@@ -768,6 +772,22 @@ def register_bootstrap_options(cls, register):
             # are likely to be able to use the same storage location.
             default=DEFAULT_LOCAL_STORE_OPTIONS.store_dir,
         )
+        register(
+            "--local-store-shard-count",
+            type=int,
+            advanced=True,
+            help=(
+                "The number of LMDB shards created for the local store."
```

Comment:
I feel like it's worth a note that running with different values on the same machine may cause significant problems...

If you go from having fewer large shards to more smaller shards, you will ?segfault pants when it starts and tries to mmap a too-large byte-slice.

If you go from having more shards to having fewer, you'll end up with some never-gc'd files.

And if you ever change the size, you'll end up what blobs in incorrect shards, which I _think_ will eventually get gc'd.

I'm not sure if it's worth some commentary around "You can experiment with this, but you should keep a consistent value once you've picked one, and also, delete ~/.cache/pants/lmdb_store any time you change it"?

Reply:
> I feel like it's worth a note that running with different values on the same machine may cause significant problems...

Could Pants detect incompatible values and then refuse to run in that case?

Reply:
Yea, some more precautions should be mentioned here.

But part of the reason this is a draft is that if we don't detect a noticeable difference with any shard count (SSDs don't need it in the two environments I tested, but want to rule out rotational drives), then we may change the lower the default and/or deprecate it.